File: project/backend/src/main/java/co/edu/utp/isc/gia/examsapp/web/dto/ExamDto.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package co.edu.utp.isc.gia.examsapp.web.dto;

import co.edu.utp.isc.gia.examsapp.web.dto.abstractdto.QuestionDto;
import java.io.Serializable;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 *
 * @author CJ
 */
@AllArgsConstructor @NoArgsConstructor
@Setter @Getter
public class ExamDto implements Serializable {
    
    private Long id;
    private String name;
    private String link;
    private Double maxGrade;
    private String description;
    private Integer examtime;
    private ProfessorDto professor;
    private List<QuestionDto> questions;
    private List<ExamStudentDto> examStudents;
}

File: project/frontend/src/app/dto/ExamDto.ts
import {OnInit } from '@angular/core';
import { ProfessorDto } from './ProfessorDto';
import { QuestionDto } from './abstractDto/QuestionDto';
import { ExamStudentDto } from './ExamStudentDto';

export class ExamDto implements OnInit {

    public id : number;
    public name : String;
    public link : String;
    public maxGrade : number;
    public description : String;
    public examtime : number;
    public professor : ProfessorDto;
    public questions : Array<QuestionDto>;
    public examStudents : Array<ExamStudentDto>;

    ngOnInit() { 
        this.id = 0;
        this.name = new String;
        this.link = new String;
        this.maxGrade = 0.0;
        this.description = new String;
        this.examtime = 0;
        this.professor = new ProfessorDto;
        this.questions = new Array<QuestionDto>();
        this.examStudents = new Array<ExamStudentDto>();
    }
}
